lib/jekyll/converters/scss.rb

```diff
@@ -286,6 +286,10 @@ def site
       def site_source
         site.source
       end
+
+      def site_source_relative_from_pwd
+        Pathname.new(site_source).relative_path_from(Pathname.new(Dir.pwd)).to_s
```

Comment:
This needs to be memoized to avoid creating new `Pathname` instances on every call.
Additionally, instead of converting the pathname to a string and then subsequently use `File.join ...`, it'd be better to leave this as a pathname and then use `Pathname#join`:
```ruby
site_source_relative_from_pwd.join(sass_page.basename + '.css')
```

Reply:
Agreed with memorizing it. However, `Pathname#join` would actually return a new instance of `Pathname`, that in the end I think won't be more efficient than `File.join`? And in the end the function is supposed to return a string anyways.

Reply:
I can confirm your thoughts. I ran some benchmarks.
`File.join` was 13x faster than `Pathname#join`